benchmarks/test_basic_parity.py

```diff
@@ -45,8 +45,8 @@ def assert_parity_absolute(pl_values, pt_values, norm_by: float = 1, max_diff: f
 @pytest.mark.parametrize(
     'cls_model,max_diff_speed,max_diff_memory',
     [
-        (ParityModuleRNN, 0.05, 0.0),
-        (ParityModuleMNIST, 0.25, 0.0),  # todo: lower this thr
+        (ParityModuleRNN, 0.05, 0.001),
+        (ParityModuleMNIST, 0.25, 0.001),  # todo: lower this thr
```

Comment:
@PyTorchLightning/core-contributors  Am I allowed to to this? 

the memory measured here in this benchmark is `torch.cuda.max_memory_allocated()`, the max cuda memory allocated during the process. This branch uses a bit more memory because when we move the batch to the device now in the new location, the batch remains in memory a bit longer than before (because we need to keep it until we call on_train_batch_end hook) and it overlaps with other memory being used by the loop, so the max has slightly increased, going over the benchmark threshold here. Let me know what you think.

Reply:
The memory difference would be back to 0 if the batch was also kept alive up until the same point in the native version, right? In that case, then it's okay with me.

Also, is 0.001 = 0.1%?

Reply:
Yes, if the vanilla loop also did all the stuff between here https://github.com/PyTorchLightning/pytorch-lightning/blob/af52de1198e9e17dcea57b4e29ad173e044fd85b/pytorch_lightning/loops/epoch/training_epoch_loop.py#L118 and L147 that we do in lightning. 

> Also, is 0.001 = 0.1%?

Yes this is a relative memory difference, measured like so:

`rel_diff = (pl_value - pt_value) / pt_value`

we then check:
`rel_diff < 0.001 = 0.1%`

Reply:
IMO this is acceptable such that we have consistency on where the batch is across module steps and callback hooks. Also this is the first time I've seen the parity trigger for memory like this. Very cool it works like that at this sensitivity!